app/code/community/Payone/Core/Model/Observer/Checkout/Onepage/Payment/Methods.php

```diff
@@ -182,8 +182,18 @@ private function restoreLastPaymentMethod(Mage_Customer_Model_Customer $customer
         if ($method) {
             $quote->getPayment()
                 ->setMethod($method)
-                ->setPayoneConfigPaymentMethodId($paymentMethodConfigId) // MAGE-395: Add Payment method id to the quote
-                ->getMethodInstance();
+                ->setPayoneConfigPaymentMethodId($paymentMethodConfigId); // MAGE-395: Add Payment method id to the quote
+
+            // MAGE-395: Get and copy stored fields for that Customer with this Method
+            /** @var Payone_Core_Model_Domain_Customer $payoneCustomer */
+            $payoneCustomer = Mage::getModel('payone_core/domain_customer');
+            $payoneCustomer = $payoneCustomer->loadByCustomerIdPaymentCode($customer->getId()+1, $method);
```

Comment:
Why `+1`?

Reply:
Thanks a lot for watching ! 
I had a little delay between implementation and commit.
this +1 was a test to simulate case of non existing data for customerId, but obviously forgotten there :/
I just pushed the correction (code removed)
Thanks again for your vigilance